File: week02_Arrays/pset/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float L = (float) count_letters(text) * 100 / count_words(text);
    float S = (float) count_sentences(text) * 100 / count_words(text);
    float ind = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(ind);

    if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Count letters
int count_letters(string text)
{
    int n = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        n = (isalpha(text[i])) ? n + 1 : n;
    }
    return n;
}
// Count words
int count_words(string text)
{
    int n = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        n = (isblank(text[i])) ? n + 1 : n;
    }
    return n + 1;
}
// Count sentences
int count_sentences(string text)
{
    int n = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            n++;
        }
    }
    return n;
}

File: week06_Python/pset/sentimental-readability/readability.py
# TODO


def main():
    text = input('Text: ')
    letters = 0
    sentences = 0
    for i in text:
        if i.isalpha():
            letters += 1
        if i == '.' or i == '!' or i == '?':
            sentences += 1
    words_list = text.split()
    words = len(words_list)

    L = letters * 100 / words
    S = sentences * 100 / words
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index > 15:
        print('Grade 16+')
    elif index < 1:
        print('Before Grade 1')
    else:
        print(f'Grade: {index}')

main()
